packages/NodeTypeResolver/tests/NodeVisitor/FunctionMethodAndClassNodeVisitor/fixtures/anonymousClass.php.inc

```diff
@@ -0,0 +1,148 @@
+                                        methodName: bar
+                                    )
+                                )
+                                attributes: array(
+                                    classNode: Stmt_Class #2
+                                    className: Rector\NodeTypeResolver\Tests\NodeVisitor\FunctionMethodAndClassNodeVisitorTest\RandomProperty\foo
+                                    methodName: bar
+                                )
+                            )
+                            1: Stmt_Expression #11(
+                                expr: Expr_FuncCall #12(
+                                    name: Name #13(
+                                        parts: array(
+                                            0: moreCode
+                                        )
+                                        attributes: array(
+                                            classNode: Stmt_Class #2
+                                            className: Rector\NodeTypeResolver\Tests\NodeVisitor\FunctionMethodAndClassNodeVisitorTest\RandomProperty\foo
+                                            methodName: baz
+                                        )
+                                    )
+                                    attributes: array(
+                                        classNode: Stmt_Class #2
+                                        className: Rector\NodeTypeResolver\Tests\NodeVisitor\FunctionMethodAndClassNodeVisitorTest\RandomProperty\foo
+                                        methodName: baz
+                                    )
+                                )
+                                attributes: array(
+                                    classNode: Stmt_Class #2
+                                    className: Rector\NodeTypeResolver\Tests\NodeVisitor\FunctionMethodAndClassNodeVisitorTest\RandomProperty\foo
+                                    methodName: baz
+                                )
+                            )
+                        )
+                        attributes: array(
+                            classNode: Stmt_Class #2
+                            className: Rector\NodeTypeResolver\Tests\NodeVisitor\FunctionMethodAndClassNodeVisitorTest\RandomProperty\foo
+                            methodName: bar
+                        )
+                    )
+                )
+                attributes: array(
+                    classNode: Stmt_Class #2
+                    className: Rector\NodeTypeResolver\Tests\NodeVisitor\FunctionMethodAndClassNodeVisitorTest\RandomProperty\foo
+                    methodName: null
+                )
+            )
+        )
+        attributes: array(
+            classNode: null
+            className: null
+            methodName: null
+        )
+    )
+)
```

Comment:
This would be hell to maintain if php-parser or Rector changes any attribute/property.

Better check this with ->getAttribute() on particular node

Reply:
That was my original plan, but it turned out that it was hard to determine which node was what and how it related to other nodes. 

Since the attributes you want rendered are whitelisted only attributes that are relevant to the visitor you want to test should be shown. I consider any changes to these attributes relevant to the test as well.

As a solution about the maintainability (and because no sane man would type this dumps themselves) you can run this command:

`UPDATE_FIXTURE=1 vendor/bin/phpunit`

It will replace the dump.

Reply:
I see. Let's give it a try then